### DISCUSSION TITLE
auto replacing string children

### DISCUSSION THREAD

**Discussion by seeden** [tamagui/tamagui#72] [Ideas]
I am moving my project from the native base to tamagui. During refactoring, I realized that you are converting string children only in the Button component. 
Native base is doing it everywhere. XStack, YStack and so on. 
The main advantage is that you can work like with standard web development and skip adding <Text> components everywhere.

Because I like this approach I created hook usePreprocessChildren and create HOC for basic layout tamagui components.

```js
import React, { useMemo, Children, type ReactNode } from 'react';
import { Trans } from '@lingui/react';
import Text from '../components/Text';
import useTrans from './useTrans';

export default function usePreprocessChildren(children: ReactNode): ReactNode {
  const t = useTrans();

  return useMemo(() => Children.map(children, (child) => {
    if (typeof child === 'string' || typeof child === 'number') {
      return (
        <Text>{child}</Text>
      );
   // added support for translation library 
    } else if (child?.type === Trans && child?.props) {
      const { id, values } = child?.props;
      if (id) {
        return (
          <Text>{t(id, values)}</Text>
        );
      }
    }

    return child;
  }), [children]);
}
```

```js YStack.tsx
function YStack(props: YStackProps, ref: any) {
  const { children , ...rest } = props;
  const updatedChildren = usePreprocessChildren(children);

  return (
    <YStackTamagui {...rest} ref={ref}>
      {updatedChildren}
    </YStackTamagui >
  );
}

export default memo(forwardRef(YStack));
```

My question is. What do you think about this idea? Will you add it for other components as well as you did for your tamagui Button component? You are using ButtonText for that case.
If not, can you tell me more of your thoughts about it? 
Once again, thank you.

**Comment by natew** [comment]
One pitfall here is that if you wrap tamagui views in a HOC like this, you lose all parent compilation optimizations (but you can get that back by using the `.extractable` helper that I don't think is documented yet (see how Button does it).

I'm not sure about the feature. It would be nice... it would also add an extra loop over every child.

There's a reason NativeBase is a lot slower! I don't think this one is it, but adding lots of features like this can get expensive.

I'll keep thinking over if this is worth it. It's possible we could have a configuration option like `autoWrapStringChildren: true` on the tamagui config that turns this on/off. Sound good?
